env0/resource_environment.go

```diff
@@ -23,27 +23,142 @@ func resourceEnvironment() *schema.Resource {
 			},
 			"name": {
 				Type:        schema.TypeString,
-				Description: "The environment's name",
+				Description: "the environment's name",
 				Required:    true,
 			},
 			"project_id": {
 				Type:        schema.TypeString,
 				Description: "project id of the environment",
 				Required:    true,
+				ForceNew:    true,
 			},
 			"template_id": {
 				Type:        schema.TypeString,
 				Description: "the template id the environment is to be created from",
 				Required:    true,
 			},
+			"workspace": {
+				Type:        schema.TypeString,
+				Description: "the terraform workspace of the environment",
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			"revision": {
+				Type:        schema.TypeString,
+				Description: "the revision the environment is to be run against",
+				Optional:    true,
+				Computed:    true,
+			},
+			"run_plan_on_pull_requests": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform plan on pull requests creations",
+				Optional:    true,
+				Default:     false,
+			},
+			"approve_plan_automatically": {
+				Type:        schema.TypeBool,
+				Description: "should deployments require manual approvals ( defaults to true )",
+				Optional:    true,
+				Default:     true,
+			},
+			"deploy_on_push": {
+				Type:        schema.TypeBool,
+				Description: "should run terraform deploy on push events",
+				Optional:    true,
+				Default:     false,
+			},
+			"auto_deploy_by_custom_glob": {
+				Type: schema.TypeString,
+				// TODO: description
+				Description: "should deploy by custom glob",
+				Optional:    true,
+			},
+			"deployment_id": {
+				Type:        schema.TypeString,
+				Description: "id of the last deployment",
+				Computed:    true,
+			},
+			"configuration": {
+				Type:        schema.TypeList,
+				Description: "terraform and environment variables for the environment",
+				Optional:    true,
+				Elem: &schema.Resource{
```

Comment:
Damn,
can't we reference the schema from `resource_configuration_variable`?

Reply:
maybe, didn't see any example for it, and it will come with extra baggage (the crud funcs) but maybe, lest talk about it

Reply:
@alonnoga I would actually try to DRY it out.
You don't have to extract it including the CRUD methods, you could simply just set the schema to some variable and used it both here and on `configuration_variable` (it's a common thing in TF providers)

Reply:
There are several differences between the schemas.
stuff that are required when creating config variables but not when including them in deployments.
I could extract individual fields but with go it will create so much fucking code that I don't want to